### PR TITLE
chore(pcd): Improving typings

### PR DESCRIPTION
**Pull request by morphoix**

### CODE REVIEW

**Review by dryabinin-actionengine — APPROVED** [review]

File: modules/pcd/src/lib/parse-pcd.ts [hunk 1]
```diff
@@ -10,15 +10,55 @@
 
 import {Schema, Field, Float32, Uint8, FixedSizeList} from '@loaders.gl/schema';
 import {getMeshBoundingBox} from '@loaders.gl/loader-utils';
-
-const LITTLE_ENDIAN = true;
-
-export default function parsePCD(data) {
+import {SchemaMetadata} from '@loaders.gl/schema/lib/schema/classes/schema';
```
Comment (ibgreen) [inline modules/pcd/src/lib/parse-pcd.ts:13]: It's not OK to import from subpaths in the src code (it is only OK to do this in the tests).

File: modules/pcd/src/lib/parse-pcd.ts [hunk 2]
```diff
@@ -10,15 +10,55 @@
...
+  count: RegExpExecArray | null | number[];
+  width: RegExpExecArray | number;
+  height: RegExpExecArray | number;
+  viewpoint: RegExpExecArray | null | string;
+  points: RegExpExecArray | number;
+  offset: {[index: string]: number};
+  rowSize: number;
+  boundingBox?: [];
+};
+type Normalized = {
```
Comment (ibgreen) [inline modules/pcd/src/lib/parse-pcd.ts:32]: NormalizedAttributes?

File: modules/pcd/src/lib/parse-pcd.ts [hunk 3]
```diff
@@ -10,15 +10,55 @@
...
+ * @param data
+ * @returns
+ */
+export default function parsePCD(data: ArrayBufferLike) {
   // parse header (always ascii format)
   const textData = new TextDecoder().decode(data);
   const pcdHeader = parsePCDHeader(textData);
 
-  let attributes;
+  let attributes: any = {};
```
Comment (ibgreen) [inline modules/pcd/src/lib/parse-pcd.ts:61]: Can we avoid any? At least do `{[attributeName: string]: any}`?

File: modules/pcd/src/lib/parse-pcd.ts [hunk 4]
```diff
@@ -58,16 +99,23 @@ export default function parsePCD(data) {
 }
 
 // Create a header that contains common data for PointCloud category loaders
-function getNormalizedHeader(PCDheader, attributes) {
-  const pointCount = PCDheader.width * PCDheader.height; // Supports "organized" point sets
-  return {
-    vertexCount: pointCount,
-    boundingBox: getMeshBoundingBox(attributes)
-  };
+function getNormalizedHeader(PCDheader: IncomingHeader, attributes: Normalized) {
```
Comment (ibgreen) [inline modules/pcd/src/lib/parse-pcd.ts:102]: Return type `Partial<PCDHeader>`?

File: modules/pcd/src/lib/parse-pcd.ts [hunk 5]
```diff
@@ -93,14 +141,21 @@ function getNormalizedAttributes(attributes) {
...
 }
 
+/**
+ * Incoming data parsing
+ * @param data
+ * @returns Header
+ */
 /* eslint-disable complexity, max-statements */
-function parsePCDHeader(data) {
+function parsePCDHeader(data: string): IncomingHeader {
```
Comment (ibgreen) [inline modules/pcd/src/lib/parse-pcd.ts:150]: Can this return type be called `PCDHeader`

File: modules/pcd/src/lib/parse-pcd.ts (hunk 3 above)
Comment (morphoix) [reply modules/pcd/src/lib/parse-pcd.ts:61]: > {[attributeName: string]: any}

In this function we will convert **attributes**  below with the function **'getNormalizedAttributes'**
so I changed **HeaderAttributes** at the top of the file like you suggested and here is like an announcement of the variable

File: modules/pcd/src/lib/parse-pcd.ts (hunk 1 above)
Comment (morphoix) [reply modules/pcd/src/lib/parse-pcd.ts]: > It's not OK to import from subpaths in the src code (it is only OK to do this in the tests).

ok, switched to local type

File: modules/pcd/src/lib/parse-pcd.ts (hunk 2 above)
Comment (morphoix) [reply modules/pcd/src/lib/parse-pcd.ts]: > NormalizedAttributes?

renamed :)

File: modules/pcd/src/lib/parse-pcd.ts (hunk 5 above)
Comment (morphoix) [reply modules/pcd/src/lib/parse-pcd.ts]: > Can this return type be called `PCDHeader`

changed

**Review by ibgreen — APPROVED** [review]

**Pull request merged**